lib/ansible/plugins/lookup/conjur_variable.py

```diff
@@ -0,0 +1,146 @@
+    description:
+      - Retrieves credentials from Conjur using the server's Conjur identity.
+    requirements:
+      - The machine running Ansible has been given a Conjur identity.
+    options:
+      _term:
+        description: Variable path
+        required: True
+"""
+
+EXAMPLES = """
+  - debug
+    msg: {{ lookup('conjur_variable', '/path/to/secret') }}
+"""
+
+RETURN = """
+  _raw:
+    description:
+      - Value stored in Conjur.
+"""
+
+import os.path
+from ansible.errors import AnsibleError, AnsibleParserError
+from ansible.plugins.lookup import LookupBase
+from base64 import b64encode
+from netrc import netrc
+from os import environ
+from time import time
+from urllib import quote_plus
+import yaml
+
+import requests
+from requests.auth import HTTPBasicAuth
+
+try:
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
+
+
+class LookupModule(LookupBase):
+
+    def run(self, terms, variables=None, **kwargs):
+
+        # Load Conjur configuration
+        conf = self.merge_dictionaries(
+            self.load_conf('/etc/conjur.conf'),
+            self.load_conf('~/.conjurrc')
+        )
+        if not conf:
+            raise AnsibleError('Conjur configuration should be in one of the following files: \'~/.conjurrc\', \'/etc/conjur.conf\'')
+        display.vvvv("configuration: {}".format(conf))
+
+        # Load Conjur identity
+        identity = self.merge_dictionaries(
+            self.load_identity('/etc/conjur.identity', conf['appliance_url']),
+            self.load_identity('~/.netrc', conf['appliance_url'])
+        )
+        if not identity:
+            raise AnsibleError('Conjur identity should be in environment variables or in one of the following paths: \'~/.netrc\', \'/etc/conjur.identity\'')
+        display.vvvv("configuration: {}".format(identity))
+
+        # Use credentials to retrieve temporary authorization token
+        conjur_url = "{}/authn/{}/{}/authenticate".format(conf['appliance_url'], conf['account'], identity['id'])
+        display.vvvv("Authentication request to Conjur at: {}, with user: {}".format(conjur_url, identity['id']))
+
+        if 'cert_file' in conf:
+            response = requests.post(conjur_url, data=identity['api_key'], verify=conf['cert_file'])
+        else:
+            response = requests.post(conjur_url, data=identity['api_key'])
+        display.vvvv("response: {}".format(response.text))
+
+        if response.status_code != 200:
+            raise AnsibleError('Failed to authenticate as \'{}\''.format(identity['id']))
+
+        # Retrieve Conjur variable using the temporary token
+        token = b64encode(response.text)
+        headers = {'Authorization': "Token token=\"{}\"".format(token)}
+        display.vvvv("Header: {}".format(headers))
+
+        url = "{}/secrets/{}/variable/{}".format(conf['appliance_url'], conf['account'], quote_plus(terms[0]))
```

Comment:
Ok, there are a bunch of these that need addressed, to index your format string for py2.6 support.